### PR TITLE
feat: load semantic headers from the second row

### PR DESCRIPTION
**Pull request by rauno56**

### CODE REVIEW

File: api/_db.data-utils.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+const filterColumnHeader = 'id';
+const certificateHeader = 'certificate';
+const nameHeader = 'name';
+const examinerHeader = 'examiner';
+const examDateHeader = 'examDate';
+const expiryDateHeader = 'expiryDate';
+const formFillTimeHeader = 'formFillTime';
+
+export const findById = (data, id) => {
+	// Drop human-readable headers
```
Comment (taavilooke) [inline api/_db.data-utils.js:77]: This comment could elaborate a bit futher on why it's done like this.

File: api/_db.data-utils.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	if (filteredRows.length > 1) {
+		console.log('Filtered more than one row', filteredRows);
+		throw new Error('More than one row');
+	}
+
+	const row = filteredRows[0];
+	const certificate = normalizeCertificate(row[certificateColumnIdx] ?? '');
+
+	const formFillDate = parseDate(row[formFillTimeColumnIdx]);
+	const expiryDate = parseDate(row[expiryDateColumnIdx]) || getExpiryTimeFromFormFillTime(formFillDate) || null;
```
Comment (taavilooke) [inline api/_db.data-utils.js:116]: What happens when neigther expiryDate or expiryTimeFromFormFillTime are set?

**Comment by taavilooke** [comment]
It seems to work correctly, but I think the row headers in the database should still be in Estonian. The point is that the first row is what ever is in the form and the second row the human readable headers. Since everything else is in Estonian, why should the headers be in English?

**Comment by rauno56** [comment]
> ... I think the row headers in the database should still be in Estonian

Aren't we going to hide the "column keys"(to distinguish them from human-readable "column headers")?

In that case, I'd say it's solely a implementation detail and "belongs" to the API, not the registry. The code is in English -> keys are in English to match the implementation.

They are just a correlation ID. I'd strongly encourage us to use the variable names in the code there, but they can technically be anything. Text with spaces and mixed casing is just more error-prone.

File: api/_db.data-utils.js (hunk 2 above)
Comment (rauno56) [reply api/_db.data-utils.js:116]: Then `null` is returned here and the UI shows it as expired.

What should happen?

**Comment by taavilooke** [comment]
My plan was to hide the first row and show the second row in the database,
because the first row will always change according to how the questions are
labeled in the forms. I guess we could also hide the second row and show
the first row for human readable headers. There are a few labels that don't
make so much sense in Estonian, but I guess we can add notes to the fields
to explain the meaning. It's better than adding a third row for the labels
and hiding the first two rows,

**Comment by taavilooke** [comment]
Hmm, I think it works as expected then. :)

File: api/_db.data-utils.js (hunk 1 above)
Comment (rauno56) [reply api/_db.data-utils.js:77]: Updated the comment.

**Comment by rauno56** [comment]
Aah... I see where you came from, now, advocating for the Estonian.

Yeah, let's definitely hide the second row and use machine-readable keys there.

1. The first one is in Estonian anyway,
2. The visible row should be something that correlates to the form or what humans talk about, thus better to be in sync with what's on form etc.

**Comment by rauno56** [comment]
@taavilooke, also let me know once the headers are changed back to match the code. We cannot merge this before.

File: api/_db.data-utils.js (hunk 2 above)
Comment (taavilooke) [reply api/_db.data-utils.js:123]: The current logic is ok.

**Review by taavilooke — APPROVED** [review]

**Pull request merged**